Add dashboard table view

somebody1234 opened the pull request:
### Pull Request Description
- Adds dummy views for all asset types (project, directory, secret, file)
  - The backend metadata does not exist for many of these columns.

### Important Notes
To do:
- [x] figure out the best way to add the custom classes - tailwind likes to avoid custom classes whenever possible

### Screenshots
![image](https://user-images.githubusercontent.com/4046547/231132353-3ef29835-5514-4fa2-bd1b-627524b655e7.png)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-electron-logic' into wip/pb/cognito-auth-authorization-app

indiv0 committed:
fix auth

indiv0 committed:
fix auth

indiv0 committed:
fix auth

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-electron-logic' into
wip/pb/cognito-auth-authorization-app

PabloBuchu committed:
PR fixes

PabloBuchu committed:
fix

PabloBuchu committed:
fix typo in project_manager

PabloBuchu committed:
remove logging

indiv0 committed:
fixes

indiv0 committed:
uninstall opener

indiv0 committed:
fixed electron handling of url

indiv0 committed:
turn object into var

indiv0 committed:
add todo

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-electron-logic

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-electron-logic

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-electron-logic

indiv0 committed:
fix style

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-electron-logic' into wip/pb/cognito-auth-authorization-app

indiv0 committed:
fixes

indiv0 committed:
remove forgot password + reset password

indiv0 committed:
remove signout

indiv0 committed:
remove setusername

indiv0 committed:
remove login

indiv0 committed:
remove registration

indiv0 committed:
fix comments

indiv0 committed:
re-enable flag

indiv0 committed:
rename div

indiv0 committed:
add comment

indiv0 committed:
fix lints

indiv0 committed:
remove tailwind conf

indiv0 committed:
Revert "remove registration"

This reverts commit 02439c9b709f5f7bdab72171d2706bec45cdfe6e.

indiv0 committed:
Revert "remove login"

This reverts commit 8e6f9c11122142c41faddef8a857cd3546d81fab.

indiv0 committed:
Revert "remove setusername"

This reverts commit 84721bcccd12b83c05f79505a5e7b444505e7917.

indiv0 committed:
Revert "remove signout"

This reverts commit 08a96d3796108f32d3f26317b01248bdfe59c5fa.

indiv0 committed:
Revert "remove forgot password + reset password"

This reverts commit e52f51a76253894041c32f9bff4e33690b354176.

indiv0 committed:
Merge branch 'develop' into wip/pb/cognito-auth-authorization-app

indiv0 committed:
remove opener

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-authorization-app' into wip/np/cognito-auth-3

indiv0 committed:
tmp

indiv0 committed:
move opener

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-authorization-app' into HEAD

indiv0 committed:
prettier

indiv0 committed:
expand docs

indiv0 committed:
tmp

indiv0 committed:
replace react-scripts with craco

indiv0 committed:
add tailwindcss

indiv0 committed:
switch to brands

indiv0 committed:
tmp

indiv0 committed:
tmp

indiv0 committed:
Merge branch 'develop' into wip/pb/cognito-auth-authorization-app

indiv0 committed:
tmp

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-authorization-app' into wip/np/cognito-auth-3

indiv0 committed:
fixmes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-3' into wip/np/cognito-auth-4

indiv0 committed:
fixmes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
fixmes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
fixmes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
fixmes

indiv0 committed:
fixes for e-hern's comments

indiv0 committed:
use abortcontroller

indiv0 committed:
add docs

indiv0 committed:
fixes

indiv0 committed:
revert craco, fix windows build

indiv0 committed:
remove from gitignore

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-authorization-app

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-authorization-app

mergify[bot] committed:
Merge branch 'develop' into wip/pb/cognito-auth-authorization-app

indiv0 committed:
remove unnecessary check

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-authorization-app' into wip/np/cognito-auth-3

indiv0 committed:
tmp

indiv0 committed:
Merge branch 'wip/pb/cognito-auth-authorization-app' into wip/np/cognito-auth-3

indiv0 committed:
augment window

indiv0 committed:
tmptmp

indiv0 committed:
split errors back up

indiv0 committed:
tmp

indiv0 committed:
tmp

indiv0 committed:
Merge branch 'develop' into wip/np/cognito-auth-3

indiv0 committed:
prettier

indiv0 committed:
fix

somebody1234 committed:
Fix lints

somebody1234 committed:
Prepare for addition for `as T` lint

somebody1234 committed:
Merge and fix lint errors

somebody1234 committed:
Add lint for early returns

somebody1234 committed:
Merge and fix lint errors

somebody1234 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

somebody1234 committed:
Address review issues

somebody1234 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

somebody1234 committed:
Fix lints

indiv0 committed:
remove withrouter

indiv0 committed:
Merge branch 'wip/np/cognito-auth-3' into wip/np/cognito-auth-4

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
Merge branch 'develop' into wip/np/cognito-auth-3

indiv0 committed:
Merge branch 'wip/np/cognito-auth-3' into wip/np/cognito-auth-4

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
fix file length

indiv0 committed:
fixes

indiv0 committed:
fixes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
remove dashboard

indiv0 committed:
remove dashboard

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
remove dashboard

indiv0 committed:
fix

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
use switch

indiv0 committed:
prettier

indiv0 committed:
fixes

indiv0 committed:
prettier

indiv0 committed:
fixes

indiv0 committed:
Merge branch 'wip/np/cognito-auth-3' into wip/np/cognito-auth-4

indiv0 committed:
run prettier

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
run prettier

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

indiv0 committed:
run prettier

indiv0 committed:
fix main page url

indiv0 committed:
allow node.js debugging

indiv0 committed:
fix lints

indiv0 committed:
change not equal

indiv0 committed:
prettier

indiv0 committed:
Merge branch 'develop' into wip/np/cognito-auth-4

indiv0 committed:
Merge branch 'wip/np/cognito-auth-4' into wip/np/cognito-auth-5

indiv0 committed:
Merge branch 'wip/np/cognito-auth-5' into wip/np/cognito-auth-6

indiv0 committed:
Merge branch 'wip/np/cognito-auth-6' into wip/np/cognito-auth-7

somebody1234 committed:
Remove references to withRouter; fix lints

somebody1234 committed:
Run prettier

somebody1234 committed:
Add cloud endpoints

somebody1234 committed:
Add JSON-RPC endpoints

somebody1234 committed:
Merge branch 'wip/somebody1234/rpc-endpoints-for-desktop-ide' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Add dashboard skeleton

somebody1234 committed:
Add dashboard skeleton

somebody1234 committed:
Add components and edit dashboard

somebody1234 commented:
other notes:
- nctdt's `projectManagerAdapter` is a good idea, so consider adding it here?
- `dashboard.tsx` is *very* long, so it may be a good idea to split it.
  - however, it's not necessarily a good idea to split it into multiple components, especially if they will only be used in `dashboard.tsx`

somebody1234 committed:
Run prettier

somebody1234 committed:
(WIP) Add cloud endpoints

somebody1234 committed:
Add rpc endpoints

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/cognito-auth-prettier

somebody1234 committed:
Address review issues

somebody1234 committed:
Formatting and minor fixes for `newtype.ts`

somebody1234 committed:
Address review issues

somebody1234 committed:
Merge remote-tracking branch 'origin/wip/somebody1234/cloud-endpoints-for-frontend' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Rename `Brand` to `NewtypeVariant`

somebody1234 committed:
Rename `Brand` to `NewtypeVariant`

somebody1234 committed:
Fix formatting in `newtype.ts`

somebody1234 committed:
Switch dashboard to esbuild

somebody1234 committed:
Minor fixes; move Tailwind generation into esbuild-config

somebody1234 committed:
Fix watching `content/` and `client/`

somebody1234 committed:
Bump esbuild binary versions; minor dependency list fixes

somebody1234 committed:
Merge branch 'wip/somebody1234/cloud-endpoints-for-frontend' into wip/somebody1234/rpc-endpoints-for-desktop-ide

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/cognito-auth-prettier

somebody1234 committed:
Run prettier

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Fixes; rename "npm run dev" to "npm run watch-dashboard"

somebody1234 committed:
Avoid writing esbuild outputs to disk for `dashboard/`

somebody1234 committed:
Convert watch-dashboard to be fully in-memory; rebuild css files on change

somebody1234 committed:
Remove obsolete FIXME

somebody1234 committed:
Remove unused constants

somebody1234 committed:
Run prettier

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

PabloBuchu committed:
add missing styles

somebody1234 committed:
Merge branch 'wip/pb/auth-fix-styling' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Fixes

somebody1234 committed:
Fix the fixes

somebody1234 committed:
Run prettier

somebody1234 committed:
Fixes; use nesting plugin to wrap tailwind preflight

somebody1234 committed:
Remove testing flag from client/watch

somebody1234 committed:
Minor fixes

somebody1234 committed:
Run prettier

somebody1234 committed:
Merge branch 'wip/somebody1234/esbuild-for-dashboard' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Export newtypes

somebody1234 committed:
Make css rebuild when tailwind config changes

somebody1234 committed:
Fix endpoints

somebody1234 committed:
Finish copying changes over

somebody1234 committed:
Remove duplicate type definitions

somebody1234 committed:
Fix bundling for dashboard

somebody1234 committed:
Fix dashboard/bundle.ts erroring when build directory does not exist

somebody1234 committed:
Move CSS to Tailwind config

somebody1234 committed:
Run prettier

somebody1234 commented:
not quite sure what the "9 new commits" thing is all about. the IDE component isn't here (as expected) though so everything should still be fine

somebody1234 commented:
or not... i merged the other PR into this one, will be fixing and force-pushing

somebody1234 committed:
Update endpoints

somebody1234 committed:
Merge branch 'wip/somebody1234/cloud-endpoints-for-frontend' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Fix esbuild binary package names

somebody1234 committed:
Remove redundant "npx" prefix from build scripts

somebody1234 committed:
Remove unused dependency

PabloBuchu committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

PabloBuchu committed:
workaround for mac freeze

PabloBuchu committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Merge branch 'wip/somebody1234/esbuild-for-dashboard' into wip/somebody1234/cognito-auth-prettier

somebody1234 committed:
Merge branch 'wip/somebody1234/cognito-auth-prettier' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Merge branch 'wip/somebody1234/cognito-auth-prettier' into wip/somebody1234/rpc-endpoints-for-desktop-ide

PabloBuchu committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

PabloBuchu committed:
add missing sections

somebody1234 committed:
Address review issue

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/esbuild-for-dashboard

somebody1234 committed:
Fix prettier config; run prettier

somebody1234 committed:
Fix live-reload of `npm run watch-dashboard`

somebody1234 committed:
Fix service worker for client-side routing

somebody1234 committed:
Remove workaround for backend bug when listing directories

PabloBuchu commented:
@somebody1234 please update with the changes from newest develop. I can't build ide on macos

PabloBuchu reviewed (changes requested):
@somebody1234 some adjustments:
- change all font sizes and weights to correspond with the figma
- the same with distances between `Drive` label and directory tree indicators and others
- can you add fixed paths and shared with so we can see how the whole path controller looks like?

somebody1234 committed:
Merge branch 'wip/somebody1234/esbuild-for-dashboard' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Merge branch 'wip/sb/prettier' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Merge branch 'wip/somebody1234/esbuild-for-dashboard' into wip/somebody1234/rpc-endpoints-for-desktop-ide

somebody1234 committed:
Merge branch 'wip/sb/prettier' into wip/somebody1234/rpc-endpoints-for-desktop-ide

somebody1234 committed:
Merge branch 'wip/somebody1234/rpc-endpoints-for-desktop-ide' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Merge branch 'wip/somebody1234/cloud-endpoints-for-frontend' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Fix sizing

somebody1234 commented:
- distances: not changed yet
- font sizes: changed
- font weights: should already be correct
- fixed paths: directory support is already there - i forgot that directories can't be added through UI yet, will add next commit
- fixed shared: added

somebody1234 committed:
Fix spacing, add fixed paths

somebody1234 commented:
all adjustments should be made now @PabloBuchu

somebody1234 committed:
Fix authentication on desktop IDE

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/rpc-endpoints-for-desktop-ide

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Allow unused locals and parameters in tsconfig.json

somebody1234 committed:
Run prettier

somebody1234 committed:
Fix TypeScript errors

somebody1234 committed:
Merge branch 'wip/somebody1234/rpc-endpoints-for-desktop-ide' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/dashboard-skeleton

somebody1234 committed:
Run prettier

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/cloud-endpoints-for-frontend

somebody1234 committed:
Merge branch 'wip/somebody1234/cloud-endpoints-for-frontend' into wip/somebody1234/dashboard-skeleton

PabloBuchu commented:
@somebody1234 are we able to open this for review?

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/dashboard-skeleton

somebody1234 commented:
@PabloBuchu this PR technically depends on #6005, however that one is *very* small so i think it might be a good idea to mark this one as ready as well.

somebody1234 commented:
#6005 should be merged into this to clean up the diff - i'll do it after this `./run backend ci-check` is done (ci-check is failing on #6212 for linux but not other OSes, for some reason)

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Merge branch 'wip/somebody1234/dashboard-skeleton' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Remove unused file

wdanilo commented:
Let's keep it as draft until its ready to merge pls :)

PabloBuchu commented:
@somebody1234 I think this one we can open for review, right?

PabloBuchu commented:
also @somebody1234  one thing from now on we should be able to see new dashboard in electron rather than standard IDE view. Can you add a flag (like `-authentication`, e.x. `-cloud-dashboard`) so we can run Enso with cloud based dashboard?

basically we don't want to hide `#dashboard` and also we won't we running `void appInstance.run()` either

PabloBuchu commented:
@somebody1234 one thing to fix. After clean `./run ide build` I am getting errors while connecting to project manager (see screenshot below). Interesting thing is that it only occur on **first** run after building the package

To reproduce:
1. Comment out `lib/content/src/index:ts:148` and `lib/content/src/index:ts:169` 
2. Build enso with: `./run ide build --skip-version-check --skip-wasm-opt --wasm-profile=dev --backend-source release --backend-release nightly`
3. Run with: `./dist/ide/mac-arm64/Enso.app/Contents/MacOS/Enso -authentication=true -debug.dev-tools`

<img width="1385" alt="Screenshot 2023-04-11 at 11 01 25" src="https://user-images.githubusercontent.com/2855109/231110746-9df6fed2-de79-45f1-934e-3dee6b697024.png">

somebody1234 commented:
- yes, will open soon after i merge
- i was assuming `-authentication` is supposed to be `-cloud-dashboard`
  - either way i was looking into it because i need a way to test changes to IDE styles so that the IDE fits inside the dashboard instead of e.g. covering it. it seems like pretty extensive changes will be required so i'm thinking of maybe delaying it for another PR, if not a completely separate PR
- re: the error, i will investigate soon

somebody1234 commented:
oh actually i'll need to add a screenshot to this before i reopen

PabloBuchu commented:
> * i was assuming `-authentication` is supposed to be `-cloud-dashboard`

At the beginning yes, but I think now we can add more granularity since the libs will start using authentication heavily but not necessarily the new dashboard (yet..)

maybe it should be a group called `cloud` with `authentication` and `dashboard`

this way you can use `-cloud.authentication=true` and `-cloud.dashboard=true`

somebody1234 commented:
and i should also probably change `#dashboard` to `.dashboard`, since the high specificity `#dashboard` makes tailwind preflight override `react-hot-toast`'s styles

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Fix eslint errors

somebody1234 committed:
Change `#dashboard` to `.dashboard` for Tailwind

somebody1234 commented:
still to do:
- add new option group
- investigate bug
- open PR

somebody1234 commented:
for posterity (since new commits have changed line numbers):
`lib/content/src/index.ts:148` is
https://github.com/enso-org/enso/blob/3de3aabd10585a20d8d4b5ca5aac275468631920/app/ide-desktop/lib/content/src/index.ts#L148

`lib/content/src/index.ts:169` is the last line here:
https://github.com/enso-org/enso/blob/3de3aabd10585a20d8d4b5ca5aac275468631920/app/ide-desktop/lib/content/src/index.ts#L167-L169

somebody1234 committed:
Split `-authentication` option into `-cloud.authentication` and `-cloud.dashboard`

PabloBuchu commented on app/ide-desktop/lib/content-config/src/config.json:
```diff
@@ -150,4 +158,4 @@
     }
   }
-}
+}
```
no nl

PabloBuchu commented on app/ide-desktop/lib/content/src/index.html:
```diff
@@ -40,7 +40,7 @@
   </head>
   <body>
-    <div id="dashboard"></div>
+    <div id="dashboard" class="dashboard"></div>
```
can we change id and class to `enso-dashboard`? then we probably need also update prefix in tailwindcss

PabloBuchu commented on app/ide-desktop/lib/dashboard/src/index.tsx:
```diff
@@ -13,13 +13,9 @@ const ESBUILD_PATH = '/esbuild'
 
 // ===================
-// === Live reload ===
+// === Live relaod ===
```
`reload`

PabloBuchu commented:
Thanks for the flags. Please address comments and open for review

somebody1234 commented:
@PabloBuchu can't repro, sorry - logs show that the project manager is being started:
```
│ Loading the window address 'http://localhost:8080?startup.platform=linux&cloud.authentication=true&engine.preferredVersion=2023.1.1-nightly.2023.4.11'.
╰ Done in 2481.7 ms.
[info] [2023-04-12T04:46:01.6Z] [org.enso.projectmanager.boot.ProjectManager$] Starting Project Manager...
[info] [2023-04-12T04:46:02.032Z] [org.enso.projectmanager.boot.ProjectManager$] Started server at 127.0.0.1:30535, press enter to kill server
```
(did a fresh rebuild just to double-check)
that said, this is using the new flags instead of commenting likes out, that might affect things

somebody1234 committed:
Address review issues

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view

somebody1234 committed:
Add description for cloud option group

PabloBuchu committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view

PabloBuchu commented:
@somebody1234 so one minor issue: `-cloud.authentication=true` doesn't show login templates without `-cloud.dashboard=true` 

listing projects from project_manager works fine, but clicking on open project calls for API. We could leave it for now and fix it in https://github.com/enso-org/enso/pull/5804 this PR

somebody1234 commented:
@PabloBuchu login templates?
if you mean the dashboard UI, that's intentional - otherwise i'm not quite sure what having a separate `dashboard` option is supposed to do
(right now, there is code to explicitly only show the dashboard if `cloud.dashboard=true`

PabloBuchu commented:
> @PabloBuchu login templates? if you mean the dashboard UI, that's intentional - otherwise i'm not quite sure what having a separate `dashboard` option is supposed to do (right now, there is code to explicitly only show the dashboard if `cloud.dashboard=true`

@somebody1234 
so when we use authentication the user should be welcomed with login page (sign in with github etc) but dashboard flag should indicate whether he lands on old IDE (appInstance.run in onAuthenticated method) or see the new dashboard with new table view introduced by this PR

somebody1234 committed:
Fix

somebody1234 commented:
@PabloBuchu a bit late but i pushed a commit that should fix it
(for posterity, it's because `DASHBOARD_PATH` is `/`, i.e. the dashboard is the default route - and it navigates away to the authentication route(s) when the user is not logged in. one of the previous commits disabled this route entirely; this one adds it back but makes it blank)

PabloBuchu committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view

PabloBuchu commented:
@somebody1234 did you pushed your latest changes? Because I still can't get to work cloud flags properly
1. `-cloud.authentication=true` should allow user for signing in and then redirect to old IDE. Instead I am getting a blank page

https://user-images.githubusercontent.com/2855109/231709224-832d42d9-2623-4c38-9fb7-f424a8071bcd.mov

2. If I was logged in previously the flag works fine and I see the old IDE

Uploading Screen Recording 2023-04-13 at 11.00.53.mov…


3. `-cloud.authentication=true -cloud.dashboard=true` looks like it works fine

https://user-images.githubusercontent.com/2855109/231709985-e8719d83-ec12-4fcd-ba53-ec3bbd12a8b7.mov

somebody1234 commented:
whoops :|
pushed now @PabloBuchu

somebody1234 committed:
Remove stray `)`

PabloBuchu commented:
Thanks works now.

QA:
- flags: 🟢 
- build & run: 🟢 
- ide watch: 🟢 
- gui watch: 🟢 

@wdanilo can you please take a look and review the code?

somebody1234 committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view

wdanilo commented on app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx:
```diff
@@ -74,14 +76,26 @@ export const SET_USERNAME_PATH = '/set-username'
     logger: loggerProvider.Logger
     platform: platformModule.Platform
+    /** Whether the dashboard should be rendered. */
+    enableDashboard: boolean
```
minor thing, but I think `showDashboard` would be a better name. `enableDashboard` makes me confused what "enabling" it really means, while, according to docs, this is only about showing it.

PabloBuchu committed:
rename enableDashbord to showDashboard

mergify[bot] committed:
Merge branch 'develop' into wip/somebody1234/dashboard-directory-view
